src/_P021_Level.ino

```diff
@@ -182,93 +389,465 @@ boolean Plugin_021(uint8_t function, struct EventStruct *event, String& string)
       String command = parseString(string, 1);
 
       if (equals(command, F("getlevel"))) {
-        string  = toString(P021_TRIGGER_LEVEL);
+        string  = toString(P021_SETPOINT);
         success = true;
       }
       else
       if (equals(command, F("gethysteresis"))) {
-        string  = toString(P021_TRIGGER_HYSTERESIS);
+        string  = toString(P021_HYSTERESIS);
         success = true;
       }
       break;
     }
 
     case PLUGIN_INIT:
     {
-      if (validGpio(CONFIG_PIN1)) {
-        pinMode(CONFIG_PIN1, OUTPUT);
+      if (validGpio(P021_GPIO_RELAY)) {
+        pinMode(P021_GPIO_RELAY, OUTPUT);
       }
       success = true;
       break;
     }
 
+    case PLUGIN_WRITE:
+    {
+      // this case defines code to be executed when the plugin executes an action (command).
+      // Commands can be accessed via rules or via http.
+      // As an example, http://192.168.1.12//control?cmd=dothis
+      // implies that there exists the comamnd "dothis"
+      // Expected commands:
+      // * levelcontrol,remote, [on|off]
+
+      // parse string to extract the command
+      String command = parseString(string, 1); // already converted to lowercase
+      String subcmd  = parseString(string, 2);
+      String value   = parseString(string, 3);
```

Comment:
Defining these variables (and a lot of others that don't change after the initial assignment) as `const` will help the compiler optimize the code.